### PR TITLE
EUI-5973 Case flag comments

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-5973

### Change description ###
Create case flag - Review flag details page - Pending Work (Part 3)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/shared/components/palette/case-flag/components/add-comments/add-comments.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
```
Comment (Daniel-Lam) [inline src/shared/components/palette/case-flag/components/add-comments/add-comments.component.ts:2]: @johnbenjamin-hmcts - I think you can also remove `EventEmitter` and `Output` from the imports, given the component no longer emits anything

File: src/shared/components/palette/case-flag/write-case-flag-field.component.spec.ts [hunk 2]
```diff
@@ -219,5 +225,27 @@ describe('WriteCaseFlagFieldComponent', () => {
     expect(component.flagsData[1].details[1].hearingRelevant).toBe(true);
   });
 
+  xit('should succeed validate and set flags case field value', () => {
```
Comment (Daniel-Lam) [inline src/shared/components/palette/case-flag/write-case-flag-field.component.spec.ts:228]: @johnbenjamin-hmcts - is this unit test still being fixed (hence the `xit`)?

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - great work... I've made a few observations but it looks good

File: src/shared/components/palette/case-flag/components/add-comments/add-comments.component.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/shared/components/palette/case-flag/components/add-comments/add-comments.component.ts:2]: Thank you @Daniel-Lam - I used the command Shift+Alt+O to correct the imports :)

File: src/shared/components/palette/case-flag/components/add-comments/add-comments.component.spec.ts [hunk 3]
```diff
@@ -24,7 +23,7 @@ xdescribe('AddCommentsComponent', () => {
     fixture = TestBed.createComponent(AddCommentsComponent);
     component = fixture.componentInstance;
     component.formGroup = new FormGroup({});
-    nextButton = fixture.debugElement.nativeElement.querySelector('button[type="button"]');
+    continueButton = fixture.debugElement.nativeElement.querySelector('button[type="button"]');
```
[the remark this answers is not in the document]
Comment (johnbenjamin-hmcts) [reply src/shared/components/palette/case-flag/components/add-comments/add-comments.component.spec.ts:26]: @Daniel-Lam - this code is not required and has been removed.

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 4]
```diff
@@ -176,18 +177,18 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
...
-              if (casefield.display_context_parameter === '#ARGUMENT(CREATE)') {
-                action = 'creation';
-              } else if (casefield.display_context_parameter === '#ARGUMENT(UPDATE)') {
-                action = 'update';
-              }
-              this.validationErrors.push({
-                id,
-                message: `Please select Next to complete the ${action} of the ${action === 'update' ? 'selected ' : ''}case flag`
-              });
+                if (this.writeCaseFlagFieldComponent.fieldState !== this.writeCaseFlagFieldComponent.caseFlagFieldState.FLAG_COMMENTS) {
```
[the remark this answers is not in the document]
Comment (Daniel-Lam) [reply src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts:180]: @johnbenjamin-hmcts - no problem :)

**Review by tarikdogan-hmcts — APPROVED** [review]

**Pull request merged**